src/main/java/org/elasticsearch/index/analysis/SkroutzGreekStemmer.java

```diff
@@ -96,6 +109,15 @@ public int stem(char s[], int len) {
     if (len < 3 || stopwords.contains(s, 0, len))
       return len;
 
+    String s_str = new String(s); //a String representation of the term's char[]
+
+    //handle step 0 and step 1 exceptions
+    char[] exceptional;
+
+    if((exceptional = step_0_exceptions.get(s_str)) != null) {
```

Comment:
I would initialise ```exceptional``` with the value from step_0_exceptions, and check its value here. It is a bit more readable this way